Expose tags in torch.ops API

anjali411 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #74512
* #72549

My original plan was to make Tags a class but I get an error that tags is not torchscriptable when I try to do that. Ultimately it might be nice to have Tags as a class and allow per-operator query as well as per-tag query

anjali411 committed:
Expose tags in torch.ops API

[ghstack-poisoned]

anjali411 committed:
Update on "Expose tags in torch.ops API"




[ghstack-poisoned]

anjali411 committed:
Update on "Expose tags in torch.ops API"


My original plan was to make Tags a class but I get an error that tags is not torchscriptable when I try to do that. Ultimately it might be nice to have Tags as a class and allow per-operator query as well as per-tag query

[ghstack-poisoned]

anjali411 committed:
Update on "Expose tags in torch.ops API"


My original plan was to make Tags a class but I get an error that tags is not torchscriptable when I try to do that. Ultimately it might be nice to have Tags as a class and allow per-operator query as well as per-tag query

[ghstack-poisoned]

anjali411 committed:
Update on "Expose tags in torch.ops API"


My original plan was to make Tags a class but I get an error that tags is not torchscriptable when I try to do that. Ultimately it might be nice to have Tags as a class and allow per-operator query as well as per-tag query

[ghstack-poisoned]

ezyang commented on torch/_ops.py:
```diff
@@ -25,6 +27,22 @@ def dl_open_guard():
+def get_tags():
+    tags = {}
+    native_yaml_path = os.path.join('aten/src/ATen/native/native_functions.yaml')
+    tags_yaml_path = os.path.join('aten/src/ATen/native/tags.yaml')
```
This isn't going to fly, pretty sure we don't ship these files in binary distribution.

ezyang commented:
Hmm, you're not going to like this, but I think the only correct way to do this is to plumb the tags into C++ (as a dispatcher level concept, similar to schema) and then use the JIT bindings to read the tags out. Otherwise, there's no way to deal with custom operators registered from C++ which may have specified tags. Can represent tags as an enum so it is not too expensive to store them binary size wise.

ezyang commented:
I think I'd probably suggest tabling this for now in favor of python operator registration api.

anjali411 committed:
Update on "Expose tags in torch.ops API"


My original plan was to make Tags a class but I get an error that tags is not torchscriptable when I try to do that. Ultimately it might be nice to have Tags as a class and allow per-operator query as well as per-tag query

[ghstack-poisoned]

anjali411 committed:
Update on "Expose tags in torch.ops API"


My original plan was to make Tags a class but I get an error that tags is not torchscriptable when I try to do that. Ultimately it might be nice to have Tags as a class and allow per-operator query as well as per-tag query

[ghstack-poisoned]

anjali411 committed:
Update on "Expose tags in torch.ops API"


My original plan was to make Tags a class but I get an error that tags is not torchscriptable when I try to do that. Ultimately it might be nice to have Tags as a class and allow per-operator query as well as per-tag query

[ghstack-poisoned]

anjali411 committed:
Update on "Expose tags in torch.ops API"


My original plan was to make Tags a class but I get an error that tags is not torchscriptable when I try to do that. Ultimately it might be nice to have Tags as a class and allow per-operator query as well as per-tag query

[ghstack-poisoned]

anjali411 committed:
Update on "Expose tags in torch.ops API"


My original plan was to make Tags a class but I get an error that tags is not torchscriptable when I try to do that. Ultimately it might be nice to have Tags as a class and allow per-operator query as well as per-tag query

[ghstack-poisoned]

anjali411 committed:
Update on "Expose tags in torch.ops API"


My original plan was to make Tags a class but I get an error that tags is not torchscriptable when I try to do that. Ultimately it might be nice to have Tags as a class and allow per-operator query as well as per-tag query

[ghstack-poisoned]
